core/src/main/java/org/apache/calcite/sql/validate/AggregatingSelectScope.java

```diff
@@ -108,19 +108,20 @@ private Resolved resolve() {
       groupExprProjection = groupAnalyzer.groupExprProjection;
     }
 
-    final Set<ImmutableBitSet> flatGroupSets =
-        Sets.newTreeSet(ImmutableBitSet.COMPARATOR);
+    final TreeMap<ImmutableBitSet, Integer> flatGroupSetCount =
```

Comment:
`SortedMap`?

Reply:
Ok, it seems `SortedMap`'s name is more intuitive.

Reply:
Isn't a regular `Map` sufficient here by the way?
It does look weird that a `SortedMap` is passed to `Map` constructor, so the order of the entries is lost.

Reply:
For the values in map, we do not care about whether they are sorted. But the flatten group sets (i.e.,the map's keyset) should be ordered when constructing Resolved.